### PR TITLE
Checkout page

### PR DESCRIPTION
**Pull request by RohitKuwar**
@Kratika0907 @sumitd94 Please review PR.
Created a generic modal component. It gets closed on clicking the cancel(cross) button, on clicking outside the modal and on clicking "Esc" key.

https://user-images.githubusercontent.com/70262783/107126146-2b13b780-68d4-11eb-8496-9703e585c546.mp4

### CODE REVIEW

File: components/Modal/index.js [hunk 1]
```diff
@@ -1,11 +1,46 @@
...
-const Modal = () => {
-  return <div> </div>;
-};
+export default function Modal({ showModal, setShowModal, render }) {
+  const modalRef = useRef();
+  const closeModal = (e) => {
+    if (modalRef.current === e.target) {
+      setShowModal(false);
+    }
+  };
```
Comment (sumitd94) [inline components/Modal/index.js:10]: I am not understanding why are we using `ref` here. Any specific reason behind it ?

Comment (RohitKuwar) [reply components/Modal/index.js:10]: @sumitd94 
Screenshot (361)
 We have given modal background as reference. So whenever modal is open and if we click on modal background (outside area of modal), it gets closed.

Comment (sumitd94) [reply components/Modal/index.js:10]: Instead let's create a backdrop and add a onclick event for closing the modal.

Comment (RohitKuwar) [reply components/Modal/index.js]: @sumitd94 Changes has been made please check.

File: components/Modal/index.js [hunk 2]
```diff
@@ -1,11 +1,41 @@
...
+    [setShowModal, showModal]
+  );
 
-Modal.propTypes = {
-  name: PropTypes.string,
-};
-
-export default Modal;
+  useEffect(() => {
+    document.addEventListener('keydown', keyPress);
```
Comment (Kratika0907) [inline components/Modal/index.js:19]: proper handler name is required. something like escKeyHandler

File: components/Modal/index.js [hunk 3]
```diff
@@ -1,11 +1,41 @@
...
+    return () => document.removeEventListener('keydown', keyPress);
+  }, [keyPress]);
+  return (
+    <>
+      {showModal ? (
+        <div>
+          <div onClick={openModal} className={classNames.backdrop}></div>
```
Comment (Kratika0907) [inline components/Modal/index.js:26]: In other file we import css as style object/namespace i.e we do styles.backdrop. Can you please check and make these consistent

File: components/cart-summary/index.js [hunk 4]
```diff
@@ -1,40 +1,73 @@
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
 import colors from '../../color/color.json';
+import Modal from '../Modal/index';
+import classNames from '../Modal/modal.module.css';
```
Comment (Kratika0907) [inline components/cart-summary/index.js:5]: Please arrange imports
Usually we have the library import first like react etc , then user module import and at last css imports

Comment (RohitKuwar) [reply components/cart-summary/index.js:6]: @Kratika0907 Changes has been made please check.

**Review by sumitd94 — APPROVED** [review]

**Review by Kratika0907 — APPROVED** [review]
Looks good to me. Thanks

**Pull request merged**